### PR TITLE
BinaryCrossEntropyLoss: added weight for positive labels

### PR DESCRIPTION
**Pull request by patrick-wilken**
I currently need weighting of positive vs. negative labels when doing a binary decision. Probably a nice to have.

### CODE REVIEW

File: returnn/tf/layers/basic.py [hunk 1]
```diff
@@ -6645,6 +6645,13 @@ class BinaryCrossEntropyLoss(Loss):
   """
   class_name = "bin_ce"
 
+  def __init__(self, pos_weight=None, **kwargs):
+    """
+    :param float pos_weight: weight of positive labels, see tf.nn.weighted_cross_entropy_with_logits.
```
Comment (albertz) [inline returnn/tf/layers/basic.py:6650]: `float|None` is the type.

File: returnn/tf/layers/basic.py [hunk 2]
```diff
@@ -6655,8 +6662,12 @@ def get_value(self):
     :rtype: tf.Tensor
     """
     with tf.name_scope("loss_bin_ce"):
-      out = tf.nn.sigmoid_cross_entropy_with_logits(
-        logits=self.output_flat, labels=tf.cast(self.target_flat, self.output_flat.dtype))
+      target_flat = tf.cast(self.target_flat, self.output_flat.dtype)
+      if self._pos_weight is None:
+        out = tf.nn.sigmoid_cross_entropy_with_logits(logits=self.output_flat, labels=target_flat)
+      else:
```
Comment (albertz) [inline returnn/tf/layers/basic.py:6668]: Do you need this branch? Wouldn't the other code just always work?

Comment (JackTemaki) [reply returnn/tf/layers/basic.py:6668]: The implementation is not exactly identical, so I think it is safer to keep it separated.

Comment (albertz) [reply returnn/tf/layers/basic.py:6668]: Ah sorry, I didn't realized this is a different function.

**Comment by patrick-wilken** [comment]
Fixed the type. And yes, I didn't want to always use the weighted loss because the formulas in the implementations look quite different, maybe there are numerical differences, just to be sure.
